### ISSUE TITLE
SetPosition and screen reader dependent visually impaired programmers

### ISSUE THREAD

**Issue by JanuszChmiel** [graemeg/fpGUI#110]
Dear MR graemeg,
Yours GUi toolkit is more than ideal solution for every visually impaired programmers. Because GUI can be created with no need to depend on Lazarus build in form designer.
But there is complex issue.
What would happen, if visually impaired programmer would not set for example The following values for AN GUI object.
    SetPosition(196, 8, 188, 16);

Can I occur severe Windows operating system related crash dialog windows? Or my app will only look unusably for The sighted majority, but screen reader could perfectly work with such GUI?
Build in GUI designer. Is it accessible with screen reader?
Thank you very much for yours advices if there is AN possibility for AN compromise. SO GUI will look atleast acceptably with no need to use object position values.

**Comment by fredvs** [comment]
Hello.

About fpGUI UIdesigner with assistive for impaired programmers, maybe you can try _designer_ext_.
It is the fpGUI UIdesigner with assistive feature too.
https://github.com/fredvs/uidesigner_ext
Binary release: https://github.com/fredvs/uidesigner_ext/releases

You may also try _ideU_, it is a IDE with assistive feature and also include _fpGUI designer_ext_ in his package.
https://github.com/fredvs/ideU/releases
Please read _README_voice_assisted.TXT_ for some instruction about voice assist.

Fre;D

**Comment by fredvs** [comment]
I agree that the _README_voice_assisted.TXT_ is not easy to understand.
It try to explain for a visually impaired programmer how to configure _ideU_ to be voice assisted.

The most easy is to ask once to somebody without view problems to run the program and then click on the "_lips_" icon.
And in menu "_Extra settings_", set "_Enable sak voice assistive at loading_" to true.
One this done, the program will be voice assistive at loading too.

It is the same for enabling voice assistive for fpGUI _uidesigner_ext_.

In any case, your comment and desire are welcome.

Fre;D

**Comment by JanuszChmiel** [comment]
There is even much more severe issue. It is related to FPGUI. Github branch. FPGUI included many examples are NVDA screen reader problematically Accessible.
SO I have been overjoyed too soon.
There is older FPGUI on SourceForge, which do not create every GUi element with different handle number. But I do not know, if it will help.
The sad fact is, that LCL apps, eventhough there are some issues many GUi elements are Accessible.
I believe, that once apone time, somebody very advanced from elite Pascal programmers group will create free units compatible witzh Lazarus and Free Pascal compiler, which will allow visually impaired programmers witzh no sight at all to create 100 % screen readers friendly GUIS including hotkeys assignment for every button, reliable focusing, ETC. But I Am aware, that doing that is complex. And especially automated positioning counting for every object is noe easy. But there is concurrent project, Autohotkey. It is written in C but there is build in mathematical formula for auto counting object positions, so object position values are not necessary.
Sure, it is interpreter, pseudocompiler, but it allow automated object position counting.

**Comment by JanuszChmiel** [comment]
JVCL XP style GUi elements can provide even hotkeys for every object. But it is depend on .lfm forms.

**Comment by fredvs** [comment]
Did you try already MSEgui and his assistive interface?
There is a demo "Coupon" that demonstrate MSEgui assistive:
https://github.com/mse-org/mseuniverse/tree/main/tools/assistive/msecoupon

**Comment by fredvs** [comment]
> NVDA screen reader problematically Accessible.

Ha, ok, you want to use NVDA.
I dont think that it will work correctly, fpGUI has his own reader via _sak_ (speech assistive kit), the same for MSEgui that use for his assistive interface _espeak_ or _espeak-ng_ voice synthesizer..

**Comment by fredvs** [comment]
> JVCL XP style GUi elements can provide even hotkeys for every object. 

With MSEgui, each elements as "shortcut" and "shortcut2" property.
